etcdserver/server.go

```diff
@@ -1275,6 +1275,8 @@ func (s *EtcdServer) applyConfChange(cc raftpb.ConfChange, confState *raftpb.Con
 // TODO: non-blocking snapshot
 func (s *EtcdServer) snapshot(snapi uint64, confState raftpb.ConfState) {
 	clone := s.store.Clone()
+	// commit kv to write metadata (for example: consistent index) to disk.
+	s.KV().Commit()
```

Comment:
snapshot happens once per 10k entries. we should make it 100k in the future. also the whole point is to make kv.commit serialized with apply. so that we wont update index (kv.commit updates consistent index internally) incorrectly.